src/js/map/MapLoading.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+
+// Required event dispatch and subscription for now while
+// parts of app are React components and others are not
+import { EventEmitter } from '../components/event-emitter';
+
+export default class MapLoading extends React.Component {
+    constructor (props) {
+        super(props);
+
+        this.state = {
+            loading: false
+        };
+    }
+
+    componentDidMount () {
+        // When a Modal component fires "on", shield turns itself on.
+        EventEmitter.subscribe('maploading:on', () => {
+            this.setState({ visible: true });
+        });
+
+        // When a Modal component fires "off", shield turns itself off.
+        EventEmitter.subscribe('maploading:off', () => {
```

Comment:
Might be good to further restrict EventEmittor names to the same class as the component that is emitting them. in this case 'maploading' is correct because MapLoading is the component name. But we don't follow that everywehre I think. This would make it very easy to lookup the file from which the event emission is coming from.